tools/ascli-actions-mdata.c

```diff
@@ -252,6 +252,50 @@ ascli_what_provides (const gchar *cachepath, const gchar *kind_str, const gchar
 	return 0;
 }
 
+/**
+ * ascli_by_categories:
+ *
+ * Get components that match the given @p categories
+ */
+int
+ascli_by_categories (const gchar *cachepath, const gchar *categories, gboolean detailed, gboolean no_cache)
+{
+    g_autoptr(AsPool) dpool = NULL;
+    g_autoptr(GPtrArray) result = NULL;
+    g_autoptr(GError) error = NULL;
+
+    if (categories == NULL) {
+        fprintf (stderr, "%s\n", _("You need to specify a categories to look up."));
+        return 2;
+    }
+
+    dpool = ascli_data_pool_new_and_open (cachepath, no_cache, &error);
+    if (error != NULL) {
+        g_printerr ("%s\n", error->message);
+        return 1;
+    }
+
+    g_auto(GStrv) cats = g_strsplit(categories, ",", 0);
+
+    result = as_pool_get_components_by_categories (dpool, cats);
+    if (result->len == 0) {
+        /* TRANSLATORS: We failed to find any component in the database, likely due to an error */
+        ascli_print_stderr (_("Unable to find components matching %s!"), categories);
```

Comment:
nit: Maybe just call it `Unable to find components in categories "%s"!` to make this more clear?